### PR TITLE
Fix timezone

### PR DESCRIPTION
**Pull request by Eslam-Nawara**
## Description
- add time flag to allow fixable time for notifications
- add timezone flag to allow different timezones

## Related issues
- #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,59 +2,59 @@
...
--   Using Docker
+- Using Docker
 
-     ```bash
-        docker build -t statusbot .
-        docker run -e <your bot token> -it statusbot
-     ```
+```bash
+docker build -t statusbot .
+docker run -e STATUSBOT_TOKEN=<your bot token> -e NOTFICATION_TIME=<notfication time> -e TIMEZONE=<timezone>-it statusbot
```
Comment (AbdelrahmanElawady) [inline README.md:41]:
```suggestion
docker run -e STATUSBOT_TOKEN=<your bot token> -e NOTFICATION_TIME=<notfication time> -e TIMEZONE=<timezone> -it statusbot
```

File: cmd/root.go [hunk 2]
```diff
@@ -40,7 +53,8 @@ func Execute() {
 }
 
 func init() {
-
 	cobra.OnInitialize()
-	rootCmd.Flags().StringP("token", "t", "", "Enter a valid telegram bot token")
+	rootCmd.Flags().StringP("bot token", "b", "", "Enter a valid telegram bot token")
```
Comment (AbdelrahmanElawady) [inline cmd/root.go:57]: flags are normally separated with hyphens so the command would look like `statusbot --bot-token`

File: internal/bot.go [hunk 3]
```diff
@@ -13,10 +14,11 @@ type Bot struct {
 	botAPI     tgbotapi.BotAPI
 	addChan    chan int64
 	removeChan chan int64
+	time       time.Time
 }
 
 // NewBot creates new bot with a valid bot api and communication channels
-func NewBot(token string) (Bot, error) {
+func NewBot(token string, t string, timezone string) (Bot, error) {
```
Comment (AbdelrahmanElawady) [inline internal/bot.go:21]: I don't understand what is `t` here.

File: internal/bot.go [hunk 4]
```diff
@@ -26,9 +28,22 @@ func NewBot(token string) (Bot, error) {
 
 	log.Printf("Authorized on account %s", botAPI.Self.UserName)
 
+	loc, err := time.LoadLocation(timezone)
+	if err != nil {
+		return bot, err
+	}
+
+	parsedTime, err := time.ParseInLocation("02 Jan 06 15:04", time.Now().Format("02 Jan 06 ")+t, loc)
```
Comment (AbdelrahmanElawady) [inline internal/bot.go:36]: I am confused on what is happening here. What is the time we are parsing and why not use the time from the input?

**Review by AbdelrahmanElawady — CHANGES_REQUESTED** [review]

File: internal/bot.go (hunk 3 above)
Comment (Eslam-Nawara) [reply internal/bot.go:21]: It's user input, time

File: internal/bot.go (hunk 4 above)
Comment (Eslam-Nawara) [reply internal/bot.go:36]: the time from the input is in format `hh:mm`, but here we try to parse date, so I add a formatted date `time.Now().Format("02 Jan 06 ")` to the input time `t` and parse them with location from the timezone

**Review by AbdelrahmanElawady — APPROVED** [review]

**Pull request merged**